### PR TITLE
Object store: get user from context

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
Retrieve user from context with support for gRPC and HTTP APIs

### CODE REVIEW

File: pkg/services/store/object/dummy/dummy_server.go [hunk 1]
```diff
@@ -47,12 +50,19 @@ func namespaceFromUID(uid string) string {
...
+	// TODO move this function outside of the `store` package
+	grpcCtx := grpccontext.FromContext(ctx)
+	if grpcCtx != nil {
+		return grpcCtx.SignedInUser
 	}
+
+	c, ok := ctxkey.Get(ctx).(*models.ReqContext)
+	if !ok || c == nil || c.SignedInUser == nil {
+		// TODO: return error?
+		return nil
```
Comment (toddtreece) [inline pkg/services/store/object/dummy/dummy_server.go:62]: seems like the caller could decide if it's an error

**Comment by ryantxu** [comment]
hymmm -- something not working right:
I created an object before applying this PR.. then updated (over HTTP with admin API key)

image

**Comment by ArturWierzbicki** [comment]
tested gRPC and it works
image

will debug http

**Comment by ArturWierzbicki** [comment]
hmm - worked for http as well!

image

**Comment by ryantxu** [comment]
I was using the "api key" interface... when I switch to service account tokens, I get:  
image

**Review by ryantxu — APPROVED** [review]
would be nice if this was written so we can use it in https://github.com/grafana/grafana/pull/56348 also :)

**Comment by ArturWierzbicki** [comment]
> would be nice if this was written so we can use it in #56348 also :)

Not sure where to put it at this point - in the `object` top-level package or somewhere accessible to anyone?

**Comment by ArturWierzbicki** [comment]
> Lets do object top level, with big comment that it should be move to global infra / new auth service structures

moved to `store/object/auth.go`

**Review by ryantxu — APPROVED** [review]
🎉

**Pull request merged**